contracts/staking/StakeManager.sol

```diff
@@ -102,16 +118,26 @@ contract StakeManager is Validator, IStakeManager, RootChainable, Lockable {
     NFTCounter = NFTCounter.add(1);
   }
 
-  function unstake(uint256 validatorId) public onlyStaker(validatorId) {
-    require(validators[validatorId].activationEpoch > 0 && validators[validatorId].deactivationEpoch == 0);
+  function unstake(uint256 validatorId) external onlyStaker(validatorId) {
+    //Todo: add state here consider jail
+    require(validators[validatorId].activationEpoch > 0 &&
+      validators[validatorId].deactivationEpoch == 0 &&
+      validators[validatorId].status == Status.Active);
+
     uint256 amount = validators[validatorId].amount;
 
-    uint256 exitEpoch = currentEpoch.add(UNSTAKE_DELAY);
+    uint256 exitEpoch = currentEpoch.add(UNSTAKE_DELAY);// notice period
```

Comment:
Something seems off here. `UNSTAKE_DELAY` is quite large, `currentEpoch` is being incremented by 1. Doesn't look like `exitEpoch` will ever reach. Same for `WITHDRAWAL_DELAY`.

Reply:
what!? 2^13 is 8k given 5 min checkpoint it is 28 days long Dynasty.
notice period would be around 50-60 days.